### PR TITLE
Fix AWS S3 Flat Directory Structure on Fetch

### PR DESCRIPTION
**Pull request by mttwise**
## Hey, I just made a Pull Request!

S3 Fetch was flattening sub-dirs into the top level directory.  This change fixes that issue by properly maintaining the original folder structure as it is defined in the S3 bucket under the key prefix that is being fetched.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hi! Thanks. We use changesets for release management and versioning, so do make one along with your PR (`yarn changeset`) and revert the modifications to the changelog and package.json.

**Comment by mttwise** [comment]
> Hi! Thanks. We use changesets for release management and versioning, so do make one along with your PR (`yarn changeset`) and revert the modifications to the changelog and package.json.

will do. will publish new revision shortly

**Comment by mttwise** [comment]
@freben I have reverted previous commits to changelog/package.json and replaced with a single changeset commit as requested

File: packages/backend-common/src/reading/tree/ReadableArrayResponse.test.ts [hunk 1]
```diff
@@ -74,9 +75,12 @@ describe('ReadableArrayResponse', () => {
   });
 
   it('should extract entire archive into directory', async () => {
+    const relativePath1 = relative(sourceDir.path, path1);
+    const relativePath2 = relative(sourceDir.path, path2);
+
     const arr: FromReadableArrayOptions = [
-      { data: createReadStream(path1), path: path1 },
-      { data: createReadStream(path2), path: path2 },
+      { data: createReadStream(path1), path: relativePath1 },
```
Comment (vinzscam) [inline packages/backend-common/src/reading/tree/ReadableArrayResponse.test.ts:82]: perhaps it could be worth adding some nested files 🤔

**Review by benjdlambert — APPROVED** [review]
Thanks for this! :tada:

I'm gonna merge this in to get it into the `next` release that's going out this afternoon, but would really appreciate it if you could do a followup PR which adds tests for the changes which would have previously failed without it.

Thinking that we missed this before because the tests weren't as exhaustive as we would have liked 👍

**Comment by mttwise** [comment]
> Thanks for this! 🎉
> 
> I'm gonna merge this in to get it into the `next` release that's going out this afternoon, but would really appreciate it if you could do a followup PR which adds tests for the changes which would have previously failed without it.
> 
> Thinking that we missed this before because the tests weren't as exhaustive as we would have liked 👍

@benjdlambert, I've updated the tests here:
https://github.com/backstage/backstage/pull/22897

**Pull request merged**